person_class.rb

```diff
@@ -1,8 +1,39 @@
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'The correct name has not been implemented yet'
+  end
+end
+
+class BaseDecorator < Nameable
+  def initialize(name_able)
+    super()
+    @name_able = name_able
+  end
+
+  def correct_name
+    @name_able.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @name_able.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
```

Comment:
- [ ] [OPTIONAL] In order to improve readability and maintainability, each class will be kept in separate files 😎